tests/unit/src/Internal/Payment/State/AbstractPaymentStateTest.php

```diff
@@ -80,9 +80,22 @@ public function test_create_state_uses_the_factory() {
 		$this->assertSame( $mock_completed_state, $result );
 	}
 
-	public function test_process_throws_exception() {
+	/**
+	 * @dataProvider state_defined_methods_provider
+	 */
+	public function test_state_methods_will_throw_exceptions( $method ) {
 		$this->expectException( StateTransitionException::class );
-		$this->expectExceptionMessage( 'The WCPay\Internal\Payment\State\AbstractPaymentState::process method is not available in the current payment state (' . PureState::class . ').' );
-		$this->sut->process( $this->createMock( PaymentRequest::class ) );
+		$this->expectExceptionMessage( 'The WCPay\Internal\Payment\State\AbstractPaymentState::' . $method . ' method is not available in the current payment state (' . PureState::class . ').' );
+
+		// Dynamically call the method based on $method.
+		$this->sut->{$method}( $this->createMock( PaymentRequest::class ) );
+	}
+
+	public function state_defined_methods_provider() {
+		return [
+			[ 'start_processing' ],
+			[ 'process' ],
```

Comment:
This one should go away, we only have `start_processing` and `complete_processing` right now.

With that in mind, let's just have 2 separate tests for the two methods. We shouldn't assume every transition method requires either the request as a parameter, or nothing.

Reply:
This is addressed.

Reply:
Good point. I missed that. Corrected here: https://github.com/Automattic/woocommerce-payments/pull/7471/commits/2c165178f28b4e6a39827c60fd9e55320d218bff